src/util/bot-runner.js

```diff
@@ -36,6 +36,11 @@ export default class BotRunner {
     }
   }
 
+  // Find the first bot for which "fn" returns true.
+  findBot(fn) {
+    return Object.keys(this.bots).map(k => this.bots[k]).find(fn);
```

Comment:
This could easily be optimized later, I wrote it to be as generic as possible for now.